src/ledctl.c

```diff
@@ -870,6 +870,99 @@ ledctl_status_code_t slot_execute(struct slot_request *slot_req)
 	}
 }
 
+bool _cmdline_parse_major_params(int opt, int opt_index, struct slot_request *req)
+{
+	switch (opt) {
+	int log_level;
+
+	case 0:
+		switch (get_option_id(longopt[opt_index].name)) {
+		case OPT_LOG_LEVEL:
+			log_level = get_option_id(optarg);
+			if (log_level != -1)
+				set_verbose_level(log_level);
+			else
+				return false;
+			break;
+		default:
+			set_verbose_level(possible_params[opt_index]);
+		}
+		break;
+	case 'l':
+		set_log_path(optarg);
+		break;
+	case 'x':
+		listed_only = 1;
+		break;
+	
+	case ':':
+	case '?':
+	default:
+		return false;
+	}
+
+	return true;
+}
+
+void _cmdline_parse_modes(int opt, struct slot_request *req)
```

Comment:
We can define arrays to describe parameters allowed only for particular mode. please look into `possible_options` array. we should generate possible options when we determined mode:
``` C
	case 'P':
		req->chosen_opt = OPT_LIST_SLOTS;
		setup_options(&longopt, &shortopt, possible_params_list_slots, possible_params_list_slots_size);
                break;
```
This will cause that only supported options will be recognized by `getopt_long()`, So if, something is not supported then error will be thrown automatically. It is your task to determine what is supported for each mode.